pennylane/operation.py

```diff
@@ -582,6 +582,41 @@ def grad_method(self):
     transforms that merge adjacent rotations.
     """
 
+    is_controlled = None
+    """bool or None: ``True`` if the operation is controlled.
+
+    If set to ``None``, the operation will be ignored during a compilation
+    transform that tries to push commuting gates through control/target wires.
+    """
+
+    target_gate_basis = None
+    """str or None: The basis of the target operation of a controlled gate.
+    If not ``None``, should take a value of ``"X"``, ``"Y"``, or ``"Z"``.
+
+    For example, ``CRX`` and ``CNOT`` have ``target_gate_basis = "X"``, whereas
+    ``ControlledPhaseShift`` and ``CZ`` have ``target_gate_basis = "Z"``.
+    """
+
+    @property
+    def comp_control_wires(self):
```

Comment:
I originally named these properties `control_wires` and `target_wires`, however this caused issues with variables of the same name used in operations like `MultiControlledX`. I prefixed with `comp_` to indicate compilation, but we should figure out a better way of incorporating this information for all gates (e.g., inheritance from a class similar to `ControlledOperation`)